### PR TITLE
Updated For ExtenX Logging

### PR DESCRIPTION
**Pull request by dkjha52**
Updated Config, TestLogger and ExtentManager files for uploading result in ExtentX.

### CODE REVIEW

File: src/main/java/com/appium/manager/TestLogger.java [hunk 1]
```diff
@@ -232,14 +232,14 @@ private void handleTestFailure(ITestResult result, String className,
...
-                    log.addScreenCaptureFromPath("screenshot/iOS/"
+                    log.addScreenCaptureFromPath(System.getProperty("user.dir") + "/target/screenshot/iOS/"
                             + device_udid.replaceAll("\\W", "_")
                             + "/" + className
                             + "/" + result.getMethod().getMethodName() + "/"
                             + screenShotNameWithTimeStamp + deviceModel + "_failed_" + result
                             .getMethod().getMethodName() + "_framed.png");
                 } else {
-                    log.addScreenCaptureFromPath("screenshot/iOS/"
+                    log.addScreenCaptureFromPath(System.getProperty("user.dir") + "/target/screenshot/iOS/"
```
Comment (saikrishna321) [inline src/main/java/com/appium/manager/TestLogger.java]: @dkjha52 can we have this relative path

**Comment by saikrishna321** [comment]
@dkjha52 can you pls fix travis

File: src/main/java/com/appium/manager/TestLogger.java [hunk 2]
```diff
@@ -232,14 +232,14 @@ private void handleTestFailure(ITestResult result, String className,
                 System.out.println("************************" + framedImageIOS.exists()
                         + "***********************");
                 if (framedImageIOS.exists()) {
-                    log.addScreenCaptureFromPath("screenshot/iOS/"
+                    log.addScreenCaptureFromPath(System.getProperty("user.dir") + "/target/screenshot/iOS/"
```
Comment (dkjha52) [inline src/main/java/com/appium/manager/TestLogger.java]: Just tried now. Using relative path, extentx is not working. The extentx is only work when we use the statement - ExtentTestManager.getTest().addScreenCaptureFromPath(System.getProperty("user.dir") + "/target/screenshot/iOS/"...... Can I update this pull request for the same.

Also note that ExtentCucumberFormatter. java contains the same statement.

**Comment by dkjha52** [comment]
Updated path to relative.

Note - 
Using relative path, extentx is not working. The extentx is only work when we use the statement - ExtentTestManager.getTest().addScreenCaptureFromPath(System.getProperty("user.dir") + "/target/screenshot/iOS/"...... Can I update this pull request for the same.
Also note that ExtentCucumberFormatter. java contains the same statement.

**Comment by saikrishna321** [comment]
@dkjha52 sure goahead and make the changes, can you also pls share the screenshot of extentX with screenshot

**Comment by dkjha52** [comment]
Please find the attached screenshots. I am going to make the changes.
screen shot cucumber
screen shot testng

**Comment by saikrishna321** [comment]
@dkjha52 thanks for the changes, these changes on the path would break someone viewing the results form remote CI machine, can you pls make changes as per this 

https://github.com/saikrishna321/AppiumTestDistribution/blob/master/src/main/java/com/appium/manager/AppiumParallelTest.java#L93-L96

**Comment by saikrishna321** [comment]
@dkjha52 ping

**Comment by dkjha52** [comment]
@saikrishna321 I tried but no luck yet. When I fix extentx then CI breaks and vice versa. I looked into the ExtentHtmlReporter. I think for fixing this issue, need htmlReporter.config().setAutoCreateRelativePathMedia(true); this method but its available in pro version only.

Can you please look into this.

**Comment by dkjha52** [comment]
If I using the below code - 
---------------
try 
					{
						log.addScreenCaptureFromPath("screenshot/browser/" + browser_udid.replaceAll("\\W", "_") + "/"
								+ className + "/" + result.getMethod().getMethodName() + "/"
								+ browserModel + "_" + result
								.getMethod().getMethodName() + "_failed.png");
					}
					catch (UncheckedIOException exception) 
					{}
---------------------
In that case screenshot is showing for CI and local but not showing in extentx. However script execution is working fine.

**Comment by saikrishna321** [comment]
@dkjha52 what if you dont use this addScreenCaptureFromPath and then just use the logger with info

**Comment by dkjha52** [comment]
@saikrishna321 If I set extentx=false then test pass and report shows the screenshot on CI as well as local.
If I set extentx=true and if any test fails then it shows exception - java.io.UncheckedIOException: java.io.IOException: The system cannot find the file specified screenshot/android/....

**Comment by dkjha52** [comment]
@saikrishna321 if I use the logger with info then its working fine. No java.io.UncheckedIOException exception is showing.

File: src/main/java/com/appium/manager/TestLogger.java [hunk 3]
```diff
@@ -207,18 +207,22 @@ private void handleTestFailure(ITestResult result, String className,
...
                                     + deviceModel + "_failed_" + result
-                                    .getMethod().getMethodName() + "_framed.png");
+                                    .getMethod().getMethodName() + "_framed.png"));
                 } else {
-                    log.addScreenCaptureFromPath(
-                            "screenshot/android/" + device_udid.replaceAll("\\W", "_") + "/"
+                    test.get().log(Status.FAIL,"Snapshot below: " 
+                        + ExtentTestManager.getTest().addScreenCaptureFromPath(
+                            System.getProperty("user.dir") + "/target/screenshot/android/" 
+                            + device_udid.replaceAll("\\W", "_") + "/"
```
Comment (AlvarezAriel) [inline src/main/java/com/appium/manager/TestLogger.java]: Why are you keeping all this repeated code and logic? This does need a refactor.

**Pull request closed without merging**